metecho/api/jobs.py

```diff
@@ -760,11 +760,14 @@ def available_task_org_config_names(epic, *, user):
             repo_name=epic.project.repo_name,
         )
         with local_github_checkout(user, repo_id) as repo_root:
+            branch_name = epic.branch_name or epic.get_base()
             config = get_project_config(
                 repo_root=repo_root,
                 repo_name=repo.name,
                 repo_url=repo.html_url,
                 repo_owner=repo.owner.login,
+                repo_branch=branch_name,
+                repo_commit=repo.branch(branch_name).latest_sha(),
```

Comment:
@jgerigmeyer This still has 2 of the problems mentioned in https://github.com/SFDO-Tooling/Metecho/pull/333:
1. local_github_checkout defaults to checking out the default branch, but then this tells CumulusCI that it has a checkout of the epic branch, which is wrong
2. Apparently there's a timing issue if we try to look up the latest commit on the epic branch shortly after it was created.

So I'd suggest:
```suggestion
            config = get_project_config(
                repo_root=repo_root,
                repo_name=repo.name,
                repo_url=repo.html_url,
                repo_owner=repo.owner.login,
                repo_branch=repo.default_branch,
                repo_commit=repo.branch(repo.default_branch).latest_sha(),
```